lib/block_view_lockups.go

```diff
@@ -0,0 +1,2178 @@
+}
+
+func (lockupLimitOperation LockupLimitOperation) ToOperationString() LockupLimitOperationString {
+	switch lockupLimitOperation {
+	case AnyLockupOperation:
+		return AnyLockupOperationString
+	case CoinLockupOperation:
+		return CoinLockupOperationString
+	case UpdateCoinLockupYieldCurveOperation:
+		return UpdateCoinLockupYieldCurveOperationString
+	case UpdateCoinLockupTransferRestrictionsOperation:
+		return UpdateCoinLockupTransferRestrictionsOperationString
+	case CoinLockupTransferOperation:
+		return CoinLockupTransferOperationString
+	case CoinLockupUnlockOperation:
+		return CoinLockupUnlockOperationString
+	default:
+		return UndefinedCoinLockupOperationString
+	}
+}
+
+func (lockupLimitOperationString LockupLimitOperationString) ToOperationType() LockupLimitOperation {
+	switch lockupLimitOperationString {
+	case AnyLockupOperationString:
+		return AnyLockupOperation
+	case CoinLockupOperationString:
+		return CoinLockupOperation
+	case UpdateCoinLockupYieldCurveOperationString:
+		return UpdateCoinLockupYieldCurveOperation
+	case UpdateCoinLockupTransferRestrictionsOperationString:
+		return UpdateCoinLockupTransferRestrictionsOperation
+	case CoinLockupTransferOperationString:
+		return CoinLockupTransferOperation
+	case CoinLockupUnlockOperationString:
+		return CoinLockupUnlockOperation
+	default:
+		return UndefinedCoinLockupOperation
+	}
+}
+
+type LockupLimitScopeType uint8
+type LockupLimitScopeTypeString string
+
+const (
+	LockupLimitScopeTypeUndefined   LockupLimitScopeType = 0
+	LockupLimitScopeTypeAnyCoins    LockupLimitScopeType = 1
+	LockupLimitScopeTypeScopedCoins LockupLimitScopeType = 2
+)
+
+const (
+	LockupLimitScopeTypeUndefinedString   LockupLimitScopeTypeString = "Undefined"
+	LockupLimitScopeTypeAnyCoinsString    LockupLimitScopeTypeString = "AnyCoins"
+	LockupLimitScopeTypeScopedCoinsString LockupLimitScopeTypeString = "ScopedCoins"
+)
+
+func (lockupLimitScopeType LockupLimitScopeType) ToString() string {
+	return string(lockupLimitScopeType.ToScopeString())
+}
+
+func (lockupLimitScopeType LockupLimitScopeType) ToScopeString() LockupLimitScopeTypeString {
+	switch lockupLimitScopeType {
+	case LockupLimitScopeTypeAnyCoins:
+		return LockupLimitScopeTypeAnyCoinsString
+	case LockupLimitScopeTypeScopedCoins:
+		return LockupLimitScopeTypeScopedCoinsString
+	default:
+		return LockupLimitScopeTypeUndefinedString
+	}
+}
+
+func (lockupLimitScopeType LockupLimitScopeTypeString) ToScopeType() LockupLimitScopeType {
+	switch lockupLimitScopeType {
+	case LockupLimitScopeTypeAnyCoinsString:
+		return LockupLimitScopeTypeAnyCoins
+	case LockupLimitScopeTypeScopedCoinsString:
+		return LockupLimitScopeTypeScopedCoins
+	default:
+		return LockupLimitScopeTypeUndefined
+	}
+}
+
+func (bav *UtxoView) _checkLockupTxnSpendingLimitAndUpdateDerivedKey(
+	derivedKeyEntry DerivedKeyEntry,
+	profilePublicKey *PublicKey,
+	lockupOperation LockupLimitOperation,
+) (DerivedKeyEntry, error) {
```

Comment:
I didn't review any of these spending limit checkers too thoroughly. Just make sure you've tested them thoroughly.